plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -983,33 +984,41 @@ func (c *PodAffinityChecker) InterPodAffinityMatches(pod *v1.Pod, meta interface
 }
 
 // anyPodMatchesPodAffinityTerm checks if any of given pods can match the specific podAffinityTerm.
-// First return value indicates whether a matching pod exists on a node that matches the topology key,
-// while the second return value indicates whether a matching pod exists anywhere.
+// First return value indicates whether enough matching pods exist on a node that matches the topology key,
+// while the second return value indicates whether enough matching pods exist anywhere.
 // TODO: Do we really need any pod matching, or all pods matching? I think the latter.
-func (c *PodAffinityChecker) anyPodMatchesPodAffinityTerm(pod *v1.Pod, allPods []*v1.Pod, node *v1.Node, term *v1.PodAffinityTerm) (bool, bool, error) {
+func (c *PodAffinityChecker) anyPodMatchesPodAffinityTerm(pod *v1.Pod, allPods []*v1.Pod, node *v1.Node, term *v1.PodAffinityTerm) (bool, bool, []error) {
 	if len(term.TopologyKey) == 0 {
-		return false, false, errors.New("Empty topologyKey is not allowed except for PreferredDuringScheduling pod anti-affinity")
+		return false, false, []error{errors.New("Empty topologyKey is not allowed except for PreferredDuringScheduling pod anti-affinity")}
 	}
-	matchingPodExists := false
+
 	namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(pod, term)
 	selector, err := metav1.LabelSelectorAsSelector(term.LabelSelector)
 	if err != nil {
-		return false, false, err
+		return false, false, []error{err}
 	}
+
+	matchingPodAnywhere := int32(1)
+	matchingPodInTopology := int32(1)
+	errs := []error{}
 	for _, existingPod := range allPods {
 		match := priorityutil.PodMatchesTermsNamespaceAndSelector(existingPod, namespaces, selector)
 		if match {
-			matchingPodExists = true
+			matchingPodAnywhere++
 			existingPodNode, err := c.info.GetNodeInfo(existingPod.Spec.NodeName)
 			if err != nil {
-				return false, matchingPodExists, err
+				errs = append(errs, err)
+				continue
 			}
 			if priorityutil.NodesHaveSameTopologyKey(node, existingPodNode, term.TopologyKey) {
-				return true, matchingPodExists, nil
+				matchingPodInTopology++
 			}
 		}
+		if matchingPodInTopology > term.MinMatchingPods && matchingPodAnywhere > term.MinMatchingPods {
+			break
```

Comment:
There're two conditions: `matchingPodInTopology > term.MinMatchingPods` && `matchingPodAnywhere > term.MinMatchingPods`; we're not sure which one or both is false; so we can not `return false, false, errs` at line 1021 :).